### PR TITLE
test: Add tests of functions in `grouping_aggregators.py`

### PR DESCRIPTION
**Pull request by BjarkeMeier**

### CODE REVIEW

File: source/databricks/calculation_engine/tests/calculation/energy/transformations/grouping_aggregators/test_aggregate_per_ga.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        actual = aggregate_per_ga(df, ANY_METERING_POINT_TYPE)
+
+        # assert
+        actual_rows = actual.df.collect()
+        assert len(actual_rows) == 1
+        actual_row = actual_rows[0]
+        assert actual_row[Colname.grid_area] == factories.DEFAULT_GRID_AREA
+        assert (
+            actual_row[Colname.sum_quantity] == 2 * factories.DEFAULT_SUM_QUANTITY
+        )  # TODO BJM: Fails due to decimal precision error in the schema of `EnergyResults`
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/calculation/energy/transformations/grouping_aggregators/test_aggregate_per_ga.py:48]: @JesperGomes, running this test line shows the problem with the `EnergyResults` as I mentioned on your PR.

File: source/databricks/calculation_engine/tests/calculation/energy/energy_results_factories.py [hunk 2]
```diff
@@ -55,16 +61,16 @@ def create_row(
         Colname.grid_area: grid_area,
         Colname.from_grid_area: from_grid_area,
         Colname.to_grid_area: to_grid_area,
-        Colname.balance_responsible_id: None,
-        Colname.energy_supplier_id: None,
+        Colname.balance_responsible_id: balance_responsible_party_id,
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/calculation/energy/energy_results_factories.py:64]: or balance_responsible_id? At least that's what the rest of the Python code use (although the official term is balance responsible party).

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/tests/calculation/energy/energy_results_factories.py]: Nice observation. I've renamed all existing occurrences (case insensitive) from `balance_responsible_party_xx` to `balance_responsible_xx`. Including `gln` to `id`.

File: source/databricks/calculation_engine/package/calculation/energy/transformations/grouping_aggregators.py [hunk 3]
```diff
@@ -33,8 +31,8 @@
 
 def aggregate_per_ga_and_brp_and_es(
     quarterly_metering_point_time_series: QuarterlyMeteringPointTimeSeries,
-    market_evaluation_point_type: MeteringPointType,
-    settlement_method: Union[SettlementMethod, None],
```
Comment (johevemi) [inline source/databricks/calculation_engine/package/calculation/energy/transformations/grouping_aggregators.py:37]: Note that | only works with python 3.10 and above

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/calculation/energy/transformations/grouping_aggregators.py:37]: We all - AFAIK - use python 3.10+ and it doesn't affect runtime (where we use 3.9.7).
But do you think that we shouldn't do this anyway?

**Review by johevemi — APPROVED** [review]

**Pull request merged**